chpl/chpl-service/src/main/java/gov/healthit/chpl/report/servicebaseurllistreport/ServiceBaseUrlListReportService.java

```diff
@@ -0,0 +1,75 @@
+package gov.healthit.chpl.report.servicebaseurllistreport;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import gov.healthit.chpl.dao.CertificationBodyDAO;
+import gov.healthit.chpl.dao.DeveloperDAO;
+import gov.healthit.chpl.domain.CertificationBody;
+import gov.healthit.chpl.domain.IdNamePair;
+import gov.healthit.chpl.scheduler.job.urluptime.UrlUptimeMonitorDAO;
+import gov.healthit.chpl.scheduler.job.urluptime.UrlUptimeMonitorTestDAO;
+
+@Component
+public class ServiceBaseUrlListReportService {
+    private UrlUptimeMonitorDAO urlUptimeMonitorDAO;
+    private UrlUptimeMonitorTestDAO urlUptimeMonitorTestDAO;
+    private DeveloperDAO developerDAO;
+    private CertificationBodyDAO certificationBodyDAO;
+
+    @Autowired
+    public ServiceBaseUrlListReportService(UrlUptimeMonitorDAO urlUptimeMonitorDAO, UrlUptimeMonitorTestDAO urlUptimeMonitorTestDAO, DeveloperDAO developerDAO,
+            CertificationBodyDAO certificationBodyDAO) {
+        this.urlUptimeMonitorDAO = urlUptimeMonitorDAO;
+        this.urlUptimeMonitorTestDAO = urlUptimeMonitorTestDAO;
+        this.developerDAO = developerDAO;
+        this.certificationBodyDAO = certificationBodyDAO;
+    }
+
+    public List<UrlUptimeMonitorEx> getUrlUptimeMonitors() {
+        List<UrlUptimeMonitorEx> urlUptimeMonitors = (List<UrlUptimeMonitorEx>) urlUptimeMonitorDAO.getAll().stream()
+                .map(monitor -> UrlUptimeMonitorEx.builder()
+                        .id(monitor.getId())
+                        .developer(monitor.getDeveloper())
+                        .url(monitor.getUrl())
+                        .datadogPublicId(monitor.getDatadogPublicId())
+                        .build())
+                .toList();
+
+
+        Map<Long, Set<CertificationBody>> developerAcbMaps = developerDAO.findAllDevelopersWithAcbs().entrySet().stream()
```

Comment:
I am not sure if this could replace these queries, but the Developer Search Service has a field for each developer "acbsForActiveListings"... wonder if this would be worth using here?